packages/core/src/lib/select/searchable-select.svelte

```diff
@@ -1,241 +1,245 @@
+};
 
-  value = option;
-  dispatch('input', value);
+const handleBlur = (event: FocusEvent) => {
+  handleSelect(autoSelectOption?.option);
+  onBlur?.(event);
 };
 
-$: {
-  if (!$isOpen && value && !options.includes(value)) {
-    value = undefined;
-    dispatch('input', value);
-  }
-}
+const handleSelect = (nextValue: string | undefined) => {
+  const fallback = exclusive && !valueInSearch ? '' : value;
+  setMenuState(CLOSED);
+  value = nextValue ?? fallback;
+  onChange?.(value);
+};
+
+const handleButtonClick = () => {
+  setMenuState(isExpanded ? CLOSED : FOCUS_SEARCH);
+  inputElement?.focus();
+};
+
+const handleKeydown = createHandleKey({
+  Enter: () => {
+    handleSelect(autoSelectOption?.option);
+  },
+  Escape: () => {
+    if (menuState === CLOSED) {
+      value = '';
+    }
+    setMenuState(CLOSED);
+  },
+  ArrowDown: (event) => {
+    if (event.altKey) {
+      setMenuState(menuState === CLOSED ? FOCUS_SEARCH : FOCUS_ITEM);
+    } else {
+      setMenuState(FOCUS_ITEM);
+      autoSelectIndex =
+        autoSelectIndex < allOptions.length - 1 ? autoSelectIndex + 1 : 0;
+    }
+  },
+  ArrowUp: () => {
+    setMenuState(FOCUS_ITEM);
+    autoSelectIndex =
+      autoSelectIndex > 0 ? autoSelectIndex - 1 : allOptions.length - 1;
+  },
+  ArrowRight: {
+    handler: () => setMenuState(FOCUS_SEARCH),
+    preventDefault: false,
+  },
+  ArrowLeft: {
+    handler: () => setMenuState(FOCUS_SEARCH),
+    preventDefault: false,
+  },
+  Home: () => {
+    setMenuState(FOCUS_SEARCH);
+    inputElement?.setSelectionRange(0, 0);
+  },
+  End: () => {
+    setMenuState(FOCUS_SEARCH);
+    inputElement?.setSelectionRange(value.length, value.length);
+  },
+});
 </script>
 
-<div
-  class={cx('relative flex h-fit w-full', extraClasses)}
-  use:clickOutside={close}
+<SelectInput
+  bind:inputElement
+  bind:value
+  {state}
+  {disabled}
+  {placeholder}
+  menuId={LIST_ID}
+  isOpen={isExpanded}
+  isFocused={menuState === FOCUS_ITEM ? false : undefined}
+  cx={[{ 'caret-transparent': menuState === FOCUS_ITEM }, inputCx]}
+  aria-autocomplete="list"
+  aria-activedescendant={activeID}
+  aria-errormessage={errorID}
+  on:focus={handleFocus}
+  on:blur={handleBlur}
+  on:keydown={handleKeydown}
+  on:input={handleInput}
+  on:click={handleButtonClick}
+/>
+<Floating
+  matchWidth
+  offset={4}
+  referenceElement={inputElement}
 >
-  <SelectInput
-    bind:value
-    isOpen={$isOpen}
-    {menuId}
-    {disabled}
-    {state}
-    {...$$restProps}
-    on:input={handleInput}
-    on:keydown={handleKeyDown}
-    on:focus={() => handleFocus(disabled)}
-    on:mousemove={() => ($isKeyboardControlling = false)}
-    on:click={() => ($isOpen ? close() : handleFocus(disabled))}
-  />
-
-  {#if !disabled}
-    <SelectMenu
-      open={$isOpen}
-      id={menuId}
-      bind:element={menu}
-      on:mouseleave={resetNavigationIndex}
-    >
-      {#if heading}
+  <ul
+    id={LIST_ID}
+    role="listbox"
+    class:hidden={!isExpanded}
```

Comment:
Went with `hidden` over `invisible` here so things wouldn't be clickable; but I'm unsure of if `invisible` was picked for some other important reason. IIRC both should be hidden from screen-readers, so I don't know of any a11y differences